ogre_mesh_viewer.py

```diff
@@ -434,20 +489,29 @@ def keyPressed(self, evt):
     def mousePressed(self, evt):
         if evt.clicks != 2:
             return True
+
         vp = self.cam.getViewport()
         ray = self.cam.getCameraToViewportRay(evt.x / vp.getActualWidth(), evt.y / vp.getActualHeight())
         self.ray_query.setRay(ray)
+        self.ray_query.setSortByDistance(True)
         for hit in self.ray_query.execute():
-            self.camman.setPivotOffset(ray.getPoint(hit.distance))
+            if evt.button == OgreBites.BUTTON_RIGHT:
+                self.camman.setPivotOffset(ray.getPoint(hit.distance))
+                return True
+
+            if hit.movable and evt.button == OgreBites.BUTTON_LEFT:
+                if hit.movable.getMovableType() == "Entity":
+                    if self.entity != None:
+                        self.entity.getParentSceneNode().showBoundingBox(False)
+
+                    self.entity = self.scn_mgr.getEntity(hit.movable.getName())
+                    self.entity.getParentSceneNode().showBoundingBox(True)
+                    self.camman.setPivotOffset(ray.getPoint(hit.distance))
```

Comment:
the camera should not move if you switch selection with a single click

Reply:
But it centers the camera on the selected object...

Reply:
imagine this happening in blender

Reply:
True, but in Blender you're not necesarily interested in inspecting the meshes which is the purpose of the mesh-viewer... that is my logic at least.

Reply:
yes, but when inspecting the scene, you do not necessary want to switch focus to a specific mesh if you e.g. just checking the transforms. For switching focus, there is the double click - which incidentally will also change selection